src/Symfony/Component/DependencyInjection/Attribute/Purpose.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * An attribute to tell how a dependency is used and hint named autowiring aliases.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class Purpose
+{
+    /**
+     * @var string
+     */
+    public $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = lcfirst(str_replace(' ', '', ucwords(preg_replace('/[^a-zA-Z0-9\x7f-\xff]++/', ' ', $name))));
```

Comment:
Are you sure about `\x7f-\xff`?

Isn't `\x80-\xff` more correct?

https://www.php.net/manual/en/language.variables.basics.php

Reply:
Did you just link Nicolas Grekas to the "Language Variables Basics" page on the PHP documentation? 😉

Reply:
This page in the doc used to contain `\x7f-\xff` for this, and the current code is full of this.
We can update, but I wouldn't in this PR, because this one just moves this line - it's not new.